src/Magento/FunctionalTestingFramework/Test/etc/mergedActionGroupSchema.xsd

```diff
@@ -15,29 +15,21 @@
         </xs:choice>
     </xs:complexType>
     <xs:complexType name="actionsRefType">
-        <xs:choice minOccurs="0" maxOccurs="unbounded">
-            <xs:group ref="actionTypeTags"/>
-            <xs:element name="arguments">
-                <xs:complexType>
-                    <xs:sequence>
-                        <xs:element name="argument" maxOccurs="unbounded" minOccurs="0">
-                            <xs:complexType>
-                                <xs:attribute type="xs:string" name="name" use="required"/>
-                                <xs:attribute type="xs:string" name="defaultValue"/>
-                                <xs:attribute type="dataTypeEnum" name="type" default="entity"/>
-                            </xs:complexType>
-                        </xs:element>
-                    </xs:sequence>
-                </xs:complexType>
-            </xs:element>
-            <xs:element name="annotations">
-                <xs:complexType>
-                    <xs:sequence>
-                        <xs:element name="description"/>
-                    </xs:sequence>
-                </xs:complexType>
-            </xs:element>
-        </xs:choice>
+        <xs:sequence>
+            <xs:choice minOccurs="0" maxOccurs="unbounded">
+                <xs:group ref="actionTypeTags"/>
+                <xs:element ref="arguments"/>
+                <xs:element ref="annotations"/>
+            </xs:choice>
+            <xs:sequence minOccurs="0">
+                <xs:group ref="returnTags"/>
+                <xs:choice minOccurs="0" maxOccurs="unbounded">
+                    <xs:group ref="actionTypeTags"/>
+                    <xs:element ref="arguments"/>
+                    <xs:element ref="annotations"/>
+                </xs:choice>
+            </xs:sequence>
+        </xs:sequence>
```

Comment:
In case `return` is forced to be the last action in an action group, this probably can be simplified.

Reply:
when `return` is forced to be the last step, it throws a schema validation error in case of merging actionGroups, based on order of reading files.

```
 <actionGroup name="MergeActionGroup">
           <arguments>
               <argument name="value" type="string"/>
        </arguments>
        <fillField selector="#foo" userInput="{{value}}" stepKey="fill1"/>
        <grabTextFrom selector="#bar" stepKey="grab1"/>
     
</actionGroup>
 <actionGroup name="MergeActionGroup">
      <fillField selector="#foo1" userInput="xyz" stepKey="fill2" after="fill1"/>
      <return value="{$grab1}" stepKey="return2"/>
</actionGroup>

```
Schema validation is run on a sort-of merged file, however, the stepKey order is not resolved at this stage -
```
<actionGroup name="MergeActionGroup" filename="/Users/sunnikri/workspace/magento2ce/app/code/Magento/AdminAnalytics/Test/Mftf/ActionGroup/MergeActionGroup.xml,/Users/sunnikri/workspace/magento2ce/app/code/Magento/Backend/Test/Mftf/ActionGroup/MergeActionGroup.xml">
      <fillField selector="#foo1" userInput="xyz" stepKey="fill2" after="fill1"/>
      <return value="{$grab1}" stepKey="return2"/>
           <arguments>
               <argument name="value" type="string"/>
        </arguments>
        <fillField selector="#foo" userInput="{{value}}" stepKey="fill1"/>
        <grabTextFrom selector="#bar" stepKey="grab1"/>
</actionGroup>
```
`return1` appears before `fill1`, and schema validation fails.

Reply:
It's fine if this cannot be simplfied.